service/dap/server.go

```diff
@@ -749,12 +755,50 @@ func (s *Server) onLaunchRequest(request *dap.LaunchRequest) {
 	// TODO(polina): Respond with an error if debug session is in progress?
 	program, ok := request.Arguments["program"].(string)
 	if !ok || program == "" {
-		s.sendErrorResponse(request.Request,
-			FailedToLaunch, "Failed to launch",
-			"The program attribute is missing in debug configuration.")
-		return
+		if !(mode == "replay") { // Only fail on modes requiring a program
```

Comment:
These nested conditions can be inlined:` if (!ok || program ="") && mode != replay`

Reply:
Done. I preserved the comment to assert the separate intent of both conditions